Branch-SDK/src/io/branch/referral/Branch.java

```diff
@@ -1096,6 +1097,18 @@ public void run() {
 	}
 
 	/**
+	 * Indicates whether or not this user has a custom identity specified for them. Note that this is independent of installs.
+	 * If you call setIdentity, this device will have that identity associated with this user until logout is called.
+	 * This includes persisting through uninstalls, as we track device id.
+	 *
+	 * @return 				A {@link Boolean} value that will return <i>true</i> only if user already has an identity.
+	 *
+	 */
+	public boolean isUserIdentified() {
+		return !prefHelper_.getIdentity().equals(PrefHelper.NO_STRING_VALUE);
```

Comment:
We don't want to have that value sent to the server, we should just have proper handling around null checks. `NO_STRING_VALUE` is just kind of a lazy hack around that. It needs to be removed throughout the SDK at some point and properly handled -- doesn't have to be this PR though.